Model/AttributeSlugRepository.php

```diff
@@ -76,8 +76,30 @@ public function __construct(
     public function save(AttributeSlugInterface $attributeSlug): AttributeSlugInterface
     {
         try {
-            /** @var AttributeSlug $attributeSlug */
-            $this->resource->save($attributeSlug);
+            //check for existing slugs with the same slug
+            try {
+                /** @var AttributeSlug $existingSlug */
+                $existingSlug = $this->findBySlug($attributeSlug->getSlug());
+
+                //slug exists, check if it is not the current attribute saved
+                if ($attributeSlug->getAttribute() != $existingSlug->getAttribute()) {
+                    $newSlug = $attributeSlug->getSlug();
+                    $newSlug .= '-';
+                    while ($newSlug === $this->findBySlug($newSlug)->getSlug()) {
+                        $newSlug .= '-';
+                    }
```

Comment:
Make use of numbers instead of constantly adding a dash. Else we can get `black-------` for example.
With a counter we get `black`, `black-1`, `black-2`. Magento does the same with url keys.     

             
    $counter = 0;
    while ($newSlug === $this->findBySlug($newSlug)->getSlug()) {
        $counter++;
        $newSlug = sprintf('%s-%s', $attributeSlug->getSlug(), $counter);
    }